indexer/etl/l1_etl.go

```diff
@@ -0,0 +1,106 @@
+package etl
+
+import (
+	"context"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	"github.com/ethereum-optimism/optimism/indexer/node"
+	"github.com/ethereum-optimism/optimism/op-e2e/config"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type L1ETL struct {
+	ETL
+
+	db *database.DB
+}
+
+func NewL1ETL(log log.Logger, db *database.DB, client node.EthClient) (*L1ETL, error) {
```

Comment:
Here you are doing literally the pattern i recommended above so same page on how to make hte l1/l2 stuff cleaner

Reply:
Generally everything is cleaner the more the code looks like
chain.pairchain
chain.l1
chain.l2
chain.origin
chain.destination
Basically the more you are treating chain objects as generic and the relationship between l1 and l2 as declarative and generic all the switching logic and room for bugs there melts away. I need to finish this library which tries to [export the pattern](https://github.com/roninjin10/rollupChains)